src/compas_fab/robots/robot.py

```diff
@@ -1360,42 +1358,52 @@ def plan_cartesian_motion(
+        ...     target_frames = [Frame([0.3, 0.1, 0.5], [1, 0, 0], [0, 1, 0]),\
                       Frame([0.5, 0.1, 0.6], [1, 0, 0], [0, 1, 0])]
+        ...     waypoints = FrameWaypoints(target_frames)
         ...     start_configuration = Configuration.from_revolute_values([-0.042, 0.033, -2.174, 5.282, -1.528, 0.000])
         ...     group = robot.main_group_name
         ...     options = {'max_step': 0.01,\
                        'jump_threshold': 1.57,\
                        'avoid_collisions': True}
-        ...     trajectory = robot.plan_cartesian_motion(frames,\
+        ...     trajectory = robot.plan_cartesian_motion(waypoints,\
                                                      start_configuration,\
                                                      group=group,\
                                                      options=options)
         ...     len(trajectory.points) > 1
         True
         """
+        # The plan_cartesian_motion method in the Robot class is a wrapper around planing backend's
+        # plan_cartesian_motion method. This method is responsible for scaling the waypoints, start_configuration and the planned trajectory.
+        # Some options may also need scaling, like max_step. This should be removed in the future.
+        # TODO: Discuss if we can have all options passed with a fixed unit to avoid scaling.
+        # Attached tools are also managed by this function.
+
         options = options or {}
-        max_step = options.get("max_step")
-        path_constraints = options.get("path_constraints")
-        attached_collision_meshes = options.get("attached_collision_meshes") or []
 
         self.ensure_client()
         if not group:
             group = self.main_group_name  # ensure semantics
 
+        # =======
+        # Scaling
+        # =======
+        need_scaling = self.scale_factor != 1.0
+
+        if need_scaling:
+            waypoints = waypoints.scaled(1.0 / self.scale_factor)
+
+        max_step = options.get("max_step")
+        if need_scaling and max_step:
+            options["max_step"] = max_step / self.scale_factor
+
         # NOTE: start_configuration has to be a full robot configuration, such
         # that all configurable joints of the whole robot are defined for planning.
         start_configuration, start_configuration_scaled = self._check_full_configuration_and_scale(start_configuration)
 
-        attached_tool = self.attached_tools.get(group)
-        if use_attached_tool_frame and attached_tool:
-            frames_WCF = self.from_tcf_to_t0cf(frames_WCF, group)
-
-        frames_WCF_scaled = []
-        for frame in frames_WCF:
-            frames_WCF_scaled.append(Frame(frame.point * 1.0 / self.scale_factor, frame.xaxis, frame.yaxis))
-
-        if path_constraints:
+        # Path constraints are only relevant to ROS Backend
+        path_constraints = options.get("path_constraints")
+        if need_scaling and path_constraints:
```

Comment:
This check is not correct. If options contains `path_constraints` but `need_scaling` ended up being false, we'll end up in the `else` block, setting ` path_constraints_WCF_scaled = None`, and basically ignoring them completely.

Reply:
I don't know if adding this boolean flag to minimize scaling operations is so worth it. It makes the program logic harder to parse for humans, and it's kind of pointless because it's not a very expensive operation to do scale with a factor of 1.0 (which is what was happening before).